samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -702,6 +703,8 @@ class SamzaContainer(
         status = SamzaContainerStatus.FAILED
     }
 
+    jmxServer.stop
```

Comment:
We are not throwing exceptions anymore. So, it doesn't matter. Either ways, it will reach Line 706. I could make it part of the shutdown sequence. But if an exception is thrown, it won't shutdown and, I need a finally. Let me know if I am missing something.